Sources/DVR/Session.swift

```diff
@@ -181,7 +186,7 @@ open class Session: URLSession {
     }
 
     private func addTask(_ task: URLSessionTask) {
-        let shouldRecord = !recording
+        let shouldRecord = !recording && (recordMode == .newEpisodes || recordMode == .once)
```

Comment:
From the descriptions in https://github.com/Azure/azure-sdk-for-ios/issues/906 I should think that if recordMode is .all it should also record.